extensions/io/org.eclipse.smarthome.io.webaudio/src/main/java/org/eclipse/smarthome/io/webaudio/internal/SSEAudioSink.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.webaudio.internal;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.audio.AudioFormat;
+import org.eclipse.smarthome.core.audio.AudioHTTPServer;
+import org.eclipse.smarthome.core.audio.AudioSink;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.eclipse.smarthome.core.audio.FixedLengthAudioStream;
+import org.eclipse.smarthome.core.audio.URLAudioStream;
+import org.eclipse.smarthome.core.audio.UnsupportedAudioFormatException;
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an audio sink that publishes an event through SSE and temporarily serves the stream via HTTP for web players
+ * to pick it up.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ */
+public class SSEAudioSink implements AudioSink {
+
+    private final Logger logger = LoggerFactory.getLogger(SSEAudioSink.class);
+
+    private static final HashSet<AudioFormat> supportedFormats = new HashSet<>();
+
+    static {
+        supportedFormats.add(AudioFormat.WAV);
+        supportedFormats.add(AudioFormat.MP3);
+    }
+
+    private AudioHTTPServer audioHTTPServer;
+
+    private EventPublisher eventPublisher;
+
+    @Override
+    public void process(AudioStream audioStream) throws UnsupportedAudioFormatException {
```

Comment:
Can you give me some details how the audio stream handling is defined for that function?
If someone passes a audio stream to a process method is the audio stream 
* closed by the method in all cases
* closed by the method if no exception is thrown
* should be closed by the caller

I don't see a consistent behaviour here.
* if it is an instance of URLAudioStream it is closed
* if it is an instance of FixedLengthAudioStream it is closed after max. 10 secnonds
* in all other cases it is not closed

As a AudioStream interface extend the InputStream interface it should be closed independent of the specific implementation.
Or miss I something here?